distributed/scheduler.py

```diff
@@ -4772,9 +4774,47 @@ def _generate_taskstates(
             # run_spec in the submitted graph may be None. This happens
             # when an already persisted future is part of the graph
             elif k in dsk:
-                # TODO run a health check to verify that run_spec and dependencies
-                # did not change. See https://github.com/dask/distributed/pull/8185
-                pass
+                try:
+                    tok_lhs: Any = tokenize(ts.run_spec, ensure_deterministic=True)
+                    tok_rhs: Any = tokenize(dsk[k], ensure_deterministic=True)
```

Comment:
Should we deal with the case where one of the two is deterministic but not the other?

Reply:
Good point

Reply:
Done